1주차/4.가계부/script.js

```diff
@@ -1,25 +1,379 @@
+outcomeModalButton.addEventListener("click", modal_outcome_button);
+function modal_income_button() {
+  incomeModalButton.style.background = "salmon";
+  outcomeModalButton.style.background = "white";
+  // 로컬 스토리지에서 수입 카테고리를 읽어옵니다.
+  const categories = JSON.parse(localStorage.getItem("categories"));
+  const incomeCategories = categories ? categories.income : [];
+  categorySelect.innerHTML = "";
+
+  // 드롭다운을 업데이트합니다.
+  categorySelect.innerHTML = incomeCategories
+    .map(
+      (category) => `
+    <option value="${category}">${category}</option>
+  `
+    )
+    .join("");
+
+  console.log(categorySelect);
+}
+function modal_outcome_button() {
+  incomeModalButton.style.background = "white";
+  outcomeModalButton.style.background = "salmon";
+  const categories = JSON.parse(localStorage.getItem("categories"));
+  const outcomeCategories = categories ? categories.outcome : [];
+  categorySelect.innerHTML = "";
+
+  // 드롭다운을 업데이트합니다.
+  categorySelect.innerHTML = outcomeCategories
+    .map(
+      (category) => `
+    <option value="${category}">${category}</option>
+  `
+    )
+    .join("");
+}
+
+function filter_history() {
+  const show_income = incomeButton.checked;
+  const show_outcome = outcomeButton.checked;
+
+  const filtered_list = filter_item(show_income, show_outcome);
+  show_list(filtered_list);
+}
+
+function filter_item(show_income, show_outcome) {
+  return HISTORY_LIST.filter((item) => {
+    if (show_income && show_outcome) {
+      return true;
+    }
+    if (show_income) return item.price > 0;
+    if (show_outcome) return item.price < 0;
+    return false;
+  });
+}
+
+const cancel_modal = document.querySelector(".cancel_modal");
+
+incomeButton.addEventListener("change", filter_history);
+outcomeButton.addEventListener("change", filter_history);
+
+let total_income = 0;
+let total_outcome = 0;
+let total_money = null;
+const total = document.querySelector(".total");
+const income = document.querySelector(".income");
+const outcome = document.querySelector(".outcome");
+function show_list(history_list) {
+  initial_list.innerHTML = ""; // 이전 내역 지우기
+  total_money = null;
+  total_income = 0;
+  total_outcome = 0;
+  const list_real = document.querySelector(".list-real");
+
+  history_list.forEach((item) => {
+    const cancelButton = document.createElement("button");
+    cancelButton.classList.add("x");
+    cancelButton.textContent = "x";
+    cancelButton.addEventListener("click", function () {
+      cancel_modal.style.display = "block";
+      const listItem = this.parentNode;
+      YesCancel.addEventListener("click", function handler() {
+        console.log(listItem);
+        const removeItem = history_list.find((x) => x.content == item.content);
+        console.log(removeItem);
+        if (removeItem.price > 0) {
+          total_income -= removeItem.price;
+        }
+        if (removeItem.price < 0) {
+          total_outcome -= removeItem.price;
+        }
```

Comment:
여기는 3항 연산자를 활용해서 이렇게 수정해볼 수 있겠다!
```suggestion
       removeItem.price > 0 ? total_income -= removeItem.price : total_outcome -= removeItem.price;
```